esmvaltool/diag_scripts/landcover/albedolandcover.py

```diff
@@ -0,0 +1,338 @@
+                    current_class = lc_classes[i_0]
+                    # First flatten the array
+                    lc_flattened = {}
+                    for varkey in current_class:
+                        lc_flattened[varkey] = model_data[varkey].data[
+                            islice, jslice].compressed()
+                    lc_sum = sum([lc_flattened[varkey]
+                                  for varkey in current_class])
+                    # Now check thresholds
+                    if (np.var(lc_sum) > 0. and
+                            len(lc_sum) >= dia_cfg['mingc']):
+                        lc_data.append(lc_sum)
+                        lc_logical[i_0] = True
+                    else:
+                        print("Variance zero or not enough\
+                               valid data for this landcover class")
+                        lc_logical[i_0] = False
+                # Now the multiple lin reg part
+                x_0 = np.stack(lc_data)
+                x_0 = x_0.swapaxes(0, 1)
+                # Same mask, so shape is fine
+                y_0 = model_data['alb'].data[islice, jslice].compressed()
+
+                # Check that the system is not over_parameterised
+                if len(y_0) > np.sum(lc_logical.astype(int)) + 1 and\
+                        np.sum(lc_logical.astype(int)) > 0:
+                    # Do multiple linear regression
+                    linreg = linear_model.LinearRegression().fit(x_0, y_0)
+                    intercept = linreg.intercept_
+                    coefficients = linreg.coef_
+
+                    # Now loop again and reconstruct albedo's
+                    lc_reg = 0
+                    for i_0 in range(3):
+                        if lc_logical[i_0]:
+                            alb_lc[i_0, i, j] = intercept\
+                                + coefficients[lc_reg] * 100.
+                            lc_reg = lc_reg + 1
+    return alb_lc
+
+
+def _write_albedochanges_to_disk(alb_lc, dia_cfg,
+                                 template_cube, datadict, cfg):
+    result_dict = {'lc1': alb_lc[0, :, :], 'lc2': alb_lc[1, :, :],
+                   'lc3': alb_lc[2, :, :]}
+    names = {'lc1': '-'.join(dia_cfg['lc1_class']),
+             'lc2': '-'.join(dia_cfg['lc2_class']),
+             'lc3': '-'.join(dia_cfg['lc3_class'])}
+    for ikey, jkey in it.product(result_dict.keys(), result_dict.keys()):
+        if not ikey == jkey:
+            print("Calculating transition from {0} to {1}".format(
+                ikey, jkey))
+            transition_name = "albedo_change_from_{0}_to_{1}".format(
+                names[ikey], names[jkey])
+            print(transition_name)
+            # Take out Frac for readability
+            transition_name = transition_name.replace('Frac', '')
+            print(transition_name)
+            transition_data = result_dict[jkey] - result_dict[ikey]
+            transition_cube = template_cube.copy(transition_data)
```

Comment:
can you not modify `template_cube` in place without copying its data to another cube?